source/containerimage/pull.go

```diff
@@ -222,7 +217,19 @@ func (p *puller) CacheKey(ctx context.Context, g session.Group, index int) (cach
 				}
 				labels["containerd.io/snapshot/remote/stargz.reference"] = p.manifest.Ref
 				labels["containerd.io/snapshot/remote/stargz.digest"] = desc.Digest.String()
-				labels["containerd.io/snapshot/remote/stargz.layers"] = layers
+				var (
+					layersKey = "containerd.io/snapshot/remote/stargz.layers"
+					layers    string
+				)
+				for _, l := range p.manifest.Remote.Descriptors[i:] {
+					ls := fmt.Sprintf("%s,", l.Digest.String())
+					// This avoids the label hits the size limitation.
+					if err := ctdlabels.Validate(layersKey, layers+ls); err != nil {
+						break
```

Comment:
Doesn't this leave the ref in unknown broken state?

Reply:
The label `containerd.io/snapshot/remote/stargz.layers` is used by stargz snapshotter to resolve layers in an image in parallel and incomplete (even if empty) list is allowed (but it possibly slows down the snapshot preparation).

When that snapshotter resolves layer specified by `containerd.io/snapshot/remote/stargz.reference` and `containerd.io/snapshot/remote/stargz.digest`, this snapshotter also [refers to layers listed in `containerd.io/snapshot/remote/stargz.layers`](https://github.com/containerd/stargz-snapshotter/blob/6ab4c0507ad44fa9d850c401849734795bea564c/stargz/fs.go#L523) and [resolves these layers as well](https://github.com/containerd/stargz-snapshotter/blob/6ab4c0507ad44fa9d850c401849734795bea564c/stargz/fs.go#L216-L245). So this snapshotter can resolve these layers before the actual `Prepare` invocations come.

Reply:
Thanks for the explanation. Can you add a comment here that skipping layers is allowed here and only affects performance.